### PR TITLE
fix: toHaveProperty with expect.any (fix: #1675)

### PR DESCRIPTION
**Pull request by skarab42**
I'm not sure I implemented the fix in the right place, but it seems to fix the problem in #1675.

### CODE REVIEW

**Comment by sheremet-va** [comment]
> @sheremet-va Does `jestEquals` support dot notation?
> 
> https://github.com/vitest-dev/vitest/blob/7c2138f2b0439dfba084f8b104af4dd628adca9b/test/core/test/jest-expect.test.ts#L199-L205

`jestEquals` compares two values. How you get them is only for you to decide.

**Comment by skarab42** [comment]
> `jestEquals` compares two values. How you get them is only for you to decide.

Is there any method already used in vitest? like [lodash.get](https://lodash.com/docs/4.17.15#get)? If not, can I use lodash? Or do I have to do my implementation? Sorry for all the questions 😅

**Comment by sheremet-va** [comment]
> > `jestEquals` compares two values. How you get them is only for you to decide.
> 
> Is there any method already used in vitest? like [lodash.get](https://lodash.com/docs/4.17.15#get)? If not, can I use lodash? Or do I have to do my implementation? Sorry for all the questions 😅

I don't think we use it anywhere. You can check source code for this matcher in chai repo, maybe they have a public API for this things.

**Comment by skarab42** [comment]
@sheremet-va I refactored using `jestEquals` and `chai.util.getPathInfo`. Thanks for pointing me to the solution ;)

File: packages/vitest/src/integrations/chai/jest-expect.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import c from 'picocolors'
-import { AssertionError } from 'chai'
+import { AssertionError, util as chaiUtil } from 'chai'
```
Comment (sheremet-va) [inline packages/vitest/src/integrations/chai/jest-expect.ts:2]: We already have utils on 

https://github.com/vitest-dev/vitest/blob/a43df9e49f2fe98bb2791d12d1c9ac3350f1ac01/packages/vitest/src/integrations/chai/jest-expect.ts#L35

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/integrations/chai/jest-expect.ts (hunk 1 above)
Comment (skarab42) [reply packages/vitest/src/integrations/chai/jest-expect.ts]: Sorry too enthusiastic, I don't look enough at what's already there... I made the change.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**